### ISSUE TITLE
GetNextPage on Playlists from search results deserializes incorrectly

### ISSUE THREAD

**Issue by one-eyed-man** [JohnnyCrazy/SpotifyAPI-NET#85]
Calling GetNextPage on the Playlists from a SearchItemsAsync call doesn't seem to deserialize correctly.

Really simple code sample to reproduce issue:

            var firstPage = await _spotify.SearchItemsAsync("punk", SearchType.Playlist, 50);

            var secondPage = await _spotify.GetNextPageAsync(firstPage.Playlists);

            if (secondPage.Items == null)
            {
                throw new NullReferenceException("Items null");
            }

![image](https://cloud.githubusercontent.com/assets/6990514/15473705/5e68d6c6-20cf-11e6-8dfc-34415be52a25.png)

`secondPage.Items` ends up being null.  Looking at the json it's trying to deserialize, it's because the SimplePlaylist model doesn't match what spotify is returning.

`{
  "playlists" : {
    "href" : "https://api.spotify.com/v1/search?query=punk&offset=50&limit=50&type=playlist",
    "items" : [ {
      "collaborative" : false,
      "external_urls" : {
        "spotify" : "http://open.spotify.com/user/billludwig/playlist/0uxgkOIkRVXe2Y8SRto7YM"
...`

Am I doing something wrong?  Or does the method need to be updated to handle this?  I'd be happy to take a crack at fixing it as long as I'm understanding this correctly.  Am I just calling the wrong method?

**Comment by JohnnyCrazy** [comment]
Ahhh yes, forgot that we also got embedded `Paging`-objects. (Same for `FullPlaylist` e.g). The current `GetNextPage`-method only works for models which are paging objects itself, like `Paging<PlaylistTrack> GetPlaylistTracks`.

Currently, I've no idea how to solve this. We may need a method in the actual model or something like that. @whereismyspaceship

**Comment by one-eyed-man** [comment]
For my own project, I just tried out solving it by deserializing the first nested member for GetNextPage instead of the root -- looks like the `playlists` member on the json result matches up with SimplePlaylist, so if i just grab it and deserialize THAT to Paging<SimplePlaylist>, I feel like it should work?  Unless I'm wrong and the models don't line up.

Other issue I'm seeing is the API is returning null for bool & int values for this search (`Collaborative`, `Public`, `Images.Height`, etc).  Not sure what to do there -- is there a way to get Newtonsoft to just make those deserialize as default(T)?  Should those members maybe be nullables?

**Comment by one-eyed-man** [comment]
For my own project, I just tried out solving it by deserializing the first nested member for GetNextPage instead of the root -- looks like the `playlists` member on the json result matches up with SimplePlaylist, so if i just grab it and deserialize THAT to Paging<SimplePlaylist>, I feel like it should work?  Unless I'm wrong and the models don't line up.

Other issue I'm seeing is the API is returning null for bool & int values for this search (`Collaborative`, `Public`, `Images.Height`, etc).  Not sure what to do there -- is there a way to get Newtonsoft to just make those deserialize as default(T)?  Should those members maybe be nullables?

**Comment by JohnnyCrazy** [comment]
Mh, I think I just found a well-working solution @one-eyed-man :

```
//Set return type specifically:
SearchItem item = _spotify.SearchItems("punk", SearchType.Playlist);
var next1 = _spotify.GetNextPage<SearchItem, SimplePlaylist>(item.Playlists);

//But also keep Old Behavior:
Paging<PlaylistTrack> tracks = _spotify.GetPlaylistTracks("artk0ree", "3POl19QMYMtxddwvLe3NZB");
var next2 = _spotify.GetNextPage(tracks);
```

Implementation:
```
public V GetNextPage<V, T>(Paging<T> paging)
{
    if (!paging.HasNextPage())
        throw new InvalidOperationException("This Paging-Object has no Next-Page");
    return DownloadData<V>(paging.Next);
}
public Paging<T> GetNextPage<T>(Paging<T> paging)
{
    return GetNextPage<Paging<T>, T>(paging);
}
```

**Comment by JohnnyCrazy** [comment]
Fixed and implemented in 8bde0643c4a9f391a53c2473b1a3e16bae34cd31

Usage:
```
Paging<PlaylistTrack> tracks = _spotify.GetPlaylistTracks("artk0ree", "3POl19QMYMtxddwvLe3NZB");
Paging<PlaylistTrack> tracks2 = _spotify.GetNextPage(tracks);

SearchItem search = _spotify.SearchItems("punk", SearchType.Playlist);
SearchItem search2 = _spotify.GetNextPage<SearchItem, SimplePlaylist>(search.Playlists);
```

**Comment by fabiangigler** [comment]
Hello friend, 


Have you read the latest article written by me? Just take a look, it's very interesting. Please read here <http://target.forthosewelove.mobi/lnkxkg>


Warmest, fgigler
